libusb/hid.c

```diff
@@ -1582,8 +1666,31 @@ int HID_API_EXPORT hid_send_feature_report(hid_device *dev, const unsigned char
 		(unsigned char *)data, length,
 		1000/*timeout millis*/);
 
-	if (res < 0)
+	if (res < 0) {
+		wchar_t *custom_string = NULL;
+
+		switch (res) {
+		case LIBUSB_ERROR_TIMEOUT:
+			custom_string = L"Transfer timed out";
+			break;
+		case LIBUSB_ERROR_PIPE:
+			custom_string = L"Control request not supported by device";
```

Comment:
that's not nesessary true

Reply:
It is what the libusb documentation says though.

```libusb_control_transfer()

Returns... LIBUSB_ERROR_PIPE if the control request was not supported by the device
```

If it's not true, then the libusb documentation should be updated.

Reply:
`ERROR_PIPE` might be returned in some other cases, not only when the device gracefully processed the request and sent a response that it is not supported. It is hard to document all corner cases, specially when some of them device-specific, etc.